### DISCUSSION TITLE
drmOpen("nvidia", NULL) returns -1

### DISCUSSION THREAD

**Discussion by YusufKhan-gamedev** [NVIDIA/open-gpu-kernel-modules#257] [Q&A] [answered]
Pretty self explanatory, why does drmOpen("nvidia", NULL) return -1 and how do I get a valid file descriptor for my nvidi a gpu?

**Comment by aritger** [comment]
Does
```
fd = open("/drv/dri/card0", O_RDWR, 0);
```
work?  (or perhaps a different path in the first argument)

**Reply by YusufKhan-gamedev** [reply]
It returns a fd of 3, running a ioctl on it returns 255. I tested /dev/dri/card1 and /dev/dri/renderD128.

**Reply by YusufKhan-gamedev** [reply]
This may need to be converted into a issue, when I run startx it complains saying "[drm] Failed to open DRM device for (null)" and "Cannot run in framebuffer mode. Please specify busIDs for all framebuffer devices".

Note: Im not running the proprietary userland.

https://paste.debian.net/1242009/

**Reply by aritger** [reply]
Just to double check: have you set the modeset=1 kernel module parameter for nvidia-drm.ko?

**Reply by YusufKhan-gamedev** [reply]
Setting it let X start up but the fd still returns garbage values

**Comment by YusufKhan-gamedev** [comment, marked as answer]
Converted to #263
